File: jsapi.py
from pyscript.js_modules import jsapi
from pyscript import document

def init_desmos():
    jsapi.init_desmos()

def set_expr(expr_id: str, expr: str):
    jsapi.set_expression(expr_id, expr)

def clear():
    jsapi.clear()

def log(text: str):
    jsapi.log(text)

def alert_user(text: str):
    jsapi.alert_user(text)

def getValueById(id: str):
    return document.getElementById(id).value

File: jsapi.js
let calculator;

export function log(text) {
    console.log(text);
}

export function initDesmos() {
    calculator = window.desmosCalculator;
}

export function setExpression(id, latex) {
    calculator.setExpression({ id, latex: latex + ' = 0' });
}

export function clear() {
    calculator.setBlank();
}

export function alertUser(msg) {
    alert(msg);
}